TutorLizard.BusinessLogic/Data/DataAccess.cs

```diff
@@ -1,5 +1,196 @@
-﻿namespace TutorLizard.BusinessLogic.Data;
+﻿using System.Text.Json;
+using TutorLizard.BusinessLogic.Models;
+
+namespace TutorLizard.BusinessLogic.Data;
 public class DataAccess : IUserIdentityDataAccess, IStudentDataAccess, ITutorDataAccess
 {
-    // This class will handle loading and saving data.
+    private List<User> _userList = new();
+    private List<ScheduleItem> _scheduleItemList = new();
+    private List<Ad> _adList = new();
+    private List<AdRequest> _adRequestList = new();
+    private List<ScheduleItemRequest> _scheduleItemRequestList = new();
+
+    private string _userFilePath = @"Data/users.json";
+    private string _scheduleItemFilePath = @"Data/schedules.json";
+    private string _adFilePath = @"Data/ads.json";
+    private string _adRequestFilePath = @"Data/ad_requests.json";
+    private string _scheduleItemRequestFilePath = @"Data/schedule_requests.json";
+
+    public DataAccess()
+    {
+        LoadData();
+    }
+
+    #region Load from Json
+    private void LoadData()
+    {
+        LoadAdsFromJson();
+        LoadSchedulesFromJson();
+        LoadUsersFromJson();
+        LoadScheduleItemRequestFromJson();
+        LoadAdRequestsFromJson();
+    }
+    private List<T> LoadFromJson<T>(string Path)
+    {
+        var filePath = $@"{Path}";
+        if (!File.Exists(filePath))
+            return new List<T>();
+
+        var jsonData = File.ReadAllText(filePath);
+
+        var outputList = JsonSerializer.Deserialize<List<T>>(jsonData, new JsonSerializerOptions
+        {
+            PropertyNameCaseInsensitive = true,
+            IncludeFields = true
+        })
+                    ?? new List<T>();
+        return outputList;
+    }
+
+    private void LoadUsersFromJson()
+    {
+        _userList = LoadFromJson<User>(_userFilePath);
+    }    
+    private void LoadAdRequestsFromJson()
+    {
+        _adRequestList = LoadFromJson<AdRequest>(_adRequestFilePath);
+    } 
+    private void LoadScheduleItemRequestFromJson()
+    {
+        _scheduleItemRequestList = LoadFromJson<ScheduleItemRequest>(_scheduleItemRequestFilePath);
+    }
+
+    private void LoadAdsFromJson()
+    {
+        _adList = LoadFromJson<Ad>(_adFilePath);
+    }
+
+    private void LoadSchedulesFromJson()
+    {
+        _scheduleItemList = LoadFromJson<ScheduleItem>(_scheduleItemFilePath);
+    }
+
+    #endregion
+
+    #region Save to Json
+    private void SaveToJson<T>(string Path, T data)
+    {
+        var filePath = $@"{Path}";
+
+        var jsonData = JsonSerializer.Serialize(data);
```

Comment:
Everything works fine, but the files are hard to read by humans, so I would suggest adding `new JsonSerializerOptions() { WriteIndented = true}` as 2nd parameter. Storage cost of formatting the files won't hurt us, but it could be helpful if we want to edit data by hand.